logstash-core/lib/logstash/runner.rb

```diff
@@ -183,7 +186,7 @@ def execute
     end
 
     LogStash::ShutdownWatcher.unsafe_shutdown = setting("pipeline.unsafe_shutdown")
-    LogStash::ShutdownWatcher.logger = @logger
+    # LogStash::ShutdownWatcher.logger = @logger
```

Comment:
Concerning logstash shutdown, it may be nice to call `LogManager.shutdown()` : this would let to flush all pending messages to logs and close nicely log files.

What you think ?